qiskit/circuit/parameterexpression.py

```diff
@@ -236,6 +236,12 @@ def __float__(self):
                             'cannot be cast to a float.'.format(self.parameters))
         return float(self._symbol_expr)
 
+    def __complex__(self):
+        if self.parameters:
+            raise TypeError('ParameterExpression with unbound parameters ({}) '
+                            'cannot be cast to a float.'.format(self.parameters))
```

Comment:
Should this be complex?
```suggestion
                            'cannot be cast to a complex.'.format(self.parameters))
```